### PR TITLE
Deprecate /_xpack/security/* in favor of /_security/*

### PR DESCRIPTION
**Pull request by jkakavas**
This commit is part of our plan to deprecate and ultimately 
remove the use of _xpack in the REST APIs. It includes:

- Adjusted REST api docs
- Adjusted HLRC docs and doc tests
- Adjusted REST actions with deprecation warnings
- Changed endpoints in rest-api-spec and relevant file names

Relates #35958

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/api/security.authenticate.json [hunk 1]
```diff
@@ -1,10 +1,12 @@
 {
```
Comment (jkakavas) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/security.authenticate.json]: I've seen the rest of the PRs are not touching the file names for these, but it made sense to me. No problem to revert if there are objections though.

**Comment by jkakavas** [comment]
`elasticsearch-ci-1` failed with an error in `testIndexFollowing` that doesn't reproduce locally
```
./gradlew :client:rest-high-level:integTestRunner -Dtests.seed=C6BDD3AC7F18F9C1 -Dtests.class=org.elasticsearch.client.CCRIT -Dtests.method="testIndexFollowing" -Dtests.security.manager=true -Dtests.locale=de -Dtests.timezone=Africa/Tunis -Dcompiler.java=11 -Druntime.java=8
```

giving it another go: 
run the gradle build tests 1

**Review by jasontedor — APPROVED** [review]
LGTM assuming that CI goes green.

**Comment by jkakavas** [comment]
@jasontedor pinging to see if you want to take a look at a509387 which adds back a number of api definitions in the deprecated format in order to use these against the old cluster in rolling upgrade tests. I couldn't think of a more elegant way to do this

**Comment by jasontedor** [comment]
@jkakavas I have a different idea, let's backport the new endpoints to 6.x (without all the other changes to the docs, specification, etc.). What do you think of this approach?

**Comment by droberts195** [comment]
The idea of backporting the new endpoints to 6.x would be consistent with what watcher is doing in #36269 (although in that case it already did support the new endpoints albeit deprecated!) and what ML is doing in #36373.  I think backporting the new endpoints is also nice because it means customers have the option to avoid deprecation warnings in the mixed version cluster during a rolling upgrade from 6.last to 7.x by switching clients to the new endpoints before starting the rolling upgrade.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by jkakavas** [comment]
@jasontedor any objections to me re-introducing ff242e826a2b49207cb6a7346703f0ce710891d8 so that we don't backport `_security` endpoints to 6.5 too for Full Cluster restart tests ?

**Comment by jkakavas** [comment]
@elasticmachine test this please

**Comment by jkakavas** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
@elasticmachine run gradle build tests 1 and run gradle build tests 2

**Comment by jkakavas** [comment]
`elasticserach-ci-tests-2` was failing because of a typo in #36379 that I corrected in 9bf0999668789c96f884bcad037fbb1063d844b6, and `elasticserach-ci-tests-1` needed additional changes for put role that was merged while this PR was fighting with CI. 

:crossed_fingers:

**Pull request merged**